src/components/device/boxfish.ts

```diff
@@ -11,8 +11,9 @@ import Locksmith from './../locksmith';
 import CameraEvents from './../../utilitis/events';
 import Detector from './../detector';
 import { EventEmitter } from 'events';
-import createBoxfishUpgraderFactory from './../upgrader/boxfishUpgraderFactory';
+import { createBoxfishUpgrader } from './../upgrader/boxfishUpgraderFactory';
 
+const MAX_UPGRADE_ATTEMT = 3;
```

Comment:
Is this legacy + hpk upgrade + hpk verify?

Reply:
No it is allowed 3 attempts if the status after upgrade is invalid.